### PR TITLE
Added PageRank Algorithm #392

### LINKED ISSUE #392: [Feature Request] PageRank Algorithm
# 🚀 Feature Request

## Is your feature request related to a problem? Please describe.
I'm frustrated when trying to understand the intricacies of the PageRank algorithm, especially its implementation and significance in web search and analysis. A clear and structured explanation would help users grasp its importance and functionality better.

## Describe the solution you'd like
I would like a comprehensive README that details the PageRank algorithm, including its definition, characteristics, time and space complexity, and a summary of its applications. This should be well-organized and easy to read for better understanding.

## Describe alternatives you've considered
I considered creating separate documents for each aspect of the PageRank algorithm, but I believe a single, concise README would be more effective for quick reference.

## Additional context
The PageRank algorithm, developed by Larry Page and Sergey Brin, is crucial for determining the importance of web pages based on their link structure. It employs a graph-based model, a random surfer model, and an iterative calculation process, with specific time and space complexity considerations. A clear explanation of these elements will enhance understanding and application in various fields, including search engines and network analysis.

## Would you like to work on this feature?
- [ ] Yes
- [ ] No

### PR DESCRIPTION
**Pull request by karthikyandrapu**
### Description
Implemented the PageRank algorithm in Python, allowing the ranking of web pages based on their link structure.

Fixes #392 

### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

### Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: docs/graphs/Page-Rank-Algorithm.md [hunk 1]
```diff
@@ -0,0 +1,107 @@
+---
+id: PageRank
```
Comment (ajay-dhangar) [inline docs/graphs/Page-Rank-Algorithm.md]: replace `id: PageRank` with `id: page-rank-algorithm`

**Review by ajay-dhangar — CHANGES_REQUESTED** [review]
update your `id ` based on suggestions

**Review by karthikyandrapu — COMMENTED** [review]
Updated id

**Review by ajay-dhangar — APPROVED** [review]
Done

**Pull request merged**